Model/Indexer/LuceneSearch.php

```diff
@@ -49,35 +52,51 @@ public function __construct(
      */
     public function prepare(): void
     {
-        $client = $this->connectionManager->getConnection();
         $value = $this->getContext()->getRequestParam('search');
-        $indexPrefix = $this->config->getIndexPrefix();
 
-        try {
-            $rawResponse = $client->query(
-                [
-                    'index' => $indexPrefix . '_async_event_*',
-                    'q' => $value,
-                    // the default page size is 10. The highest limit is 10000. If we want to traverse further, we will
-                    // have to use the search after parameter. There are no plans to implement this right now.
-                    'size' => 100
-                ]
-            );
+        if ($this->asyncEventsConfig->isIndexingEnabled()) {
+            $client = $this->connectionManager->getConnection();
+            $indexPrefix = $this->config->getIndexPrefix();
+            $filter = $this->filterBuilder->setConditionType('in')
+                ->setField($this->getName());
 
-            $rawDocuments = $rawResponse['hits']['hits'] ?? [];
-            $asyncEventIds = array_column($rawDocuments, '_id');
+            if ($value === "") {
+                return;
+            }
+
+            try {
+                $rawResponse = $client->query(
+                    [
+                        'index' => $indexPrefix . '_async_event_*',
+                        'q' => $value,
+                        // the default page size is 10. The highest limit is 10000. If we want to traverse further, we
+                        // will  have to use the search after parameter. There are no plans to implement this right now.
+                        'size' => 100
+                    ]
+                );
+
+                $rawDocuments = $rawResponse['hits']['hits'] ?? [];
+                $asyncEventIds = array_column($rawDocuments, '_id');
+
+                if (!empty($asyncEventIds)) {
+                    $filter->setValue($asyncEventIds);
+                } else {
+                    $filter->setValue("0");
+                }
+            } catch (Exception) {
+                // If we're unable to connect to Elasticsearch, we'll return nothing
+                $filter->setValue("0");
+            }
 
-            if (!empty($asyncEventIds)) {
-                $filter = $this->filterBuilder->setConditionType('in')
-                    ->setField($this->getName())
-                    ->setValue($asyncEventIds)
-                    ->create();
+            $this->getContext()->getDataProvider()->addFilter($filter->create());
+        } else {
+            if ((string)$value !== '') {
```

Comment:
`$value` is a request parameter, which means that in theory, it could be an array (e.g. `https://store.com/results?search[]=123&search[]=456`).
Now, I'm assuming this will rarely (if ever) happen, but just a little wary of the practice.
Could this be `if (!empty($value))` instead?